generator/transformers/common_producer.py

```diff
@@ -199,6 +199,10 @@ def extract_constructor(self, data: list, mandatory: bool) -> dict:
         arguments = [self.argument_named(origin=first.origin, constructor_argument=self.parentheses(first),
                                          variable=first.constructor_argument, deprecated=first.deprecated)]
         for param in data:
+            if param.deprecated:
```

Comment:
`SeatControlData.id` is still in the constructor

Reply:
Should be fixed. Since `id` was the first item in the list of mandatory parameters, it was getting added in the initializer for the argument list.